File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
from collections import Counter 

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        ans=[]
        lms=Counter(nums)
        common = lms.most_common(k)
        for i in common:
            ans.append(i[0])
        return ans
                
        
        

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int product = 1;
        int count = 0;

        for (auto i : nums){
            if (i!=0){
                product *= i;
            }else {
                count++;
            }
        }

        if (count > 1){
            return vector<int>(nums.size(),0);
        }

        vector<int> ans;

        for(auto i : nums){
            if (count > 0){
                if (i==0){
                    ans.push_back(product);
                    
                }else{
                    ans.push_back(0);
                }
                continue;
            }
            ans.push_back(product/i);
        }

        return ans;
    }
};
